tensorflow/contrib/cmake/tf_python.cmake

```diff
@@ -597,6 +592,55 @@ add_library(pywrap_tensorflow_internal SHARED
     "${tensorflow_source_dir}/tensorflow/c/tf_status_helper.cc"
     "${tensorflow_source_dir}/tensorflow/c/tf_status_helper.h"
     "${CMAKE_CURRENT_BINARY_DIR}/pywrap_tensorflow_internal.cc"
+)
+
+if(WIN32)
+    # Windows: build a static library with the same objects as tensorflow.dll.
+    # This can be used to build for a standalone exe and also helps us to
+    # find all symbols that need to be exported from the dll which is needed
+    # to provide the tensorflow c/c++ api in tensorflow.dll.
+    # From the static library we create the def file with all symbols that need to
+    # be exported from tensorflow.dll. Because there is a limit of 64K sybmols
+    # that can be exported, we filter the symbols with a python script to the namespaces
+    # we need.
+    #
```

Comment:
Just so I've got the process correct in my head: we're building a static `.lib` in order to get something that we can feed to `dumpbin` and then a custom Python script, ultimately to get a `.def` file. Then we throw away the `.lib` and pass the `.def` to a second `link` command, along with all the same objects, and we get a `.dll` with the appropriate exports?